builtin/logical/pki/path_roles.go

```diff
@@ -57,13 +57,11 @@ name in a request`,
 			},
 
 			"allowed_domains": &framework.FieldSchema{
-				Type:    framework.TypeString,
-				Default: "",
+				Type: framework.TypeCommaStringSlice,
 				Description: `If set, clients can request certificates for
 subdomains directly beneath these domains, including
 the wildcard subdomains. See the documentation for more
-information. This parameter accepts a comma-separated list
-of domains.`,
```

Comment:
Rather than remove the notification that it can accept a comma-separated list, I'd just add that it can accept that or an array. Makes things more explicit.